docs/user-guide/topics/ab_testing.md

```diff
@@ -0,0 +1,203 @@
+- All objects in the baseline versßion are named using only the application name
+- All objects in a candidate version are named as <application_name>-candidate-<index> where index is 1, 2, etc.
+- The baseline track label is the name of the application
+- Candidate track labels are of the form <application_name>-candidate-<index>
+
+## Developing Frontends: Using the SDK
+
+The basic steps to author a frontend application component using the Iter8 SDK are outlined below for *Node.js* and *Go*. Similar steps would be required for any gRPC supported langauge.
+
+### Use/Import language specific libraries
+
+The gRPC protocol buffer definition is used to generate language specific implementation. These files can be used directly or packaged and imported as a library. As examples, the Node.js sample uses manually generated files directly. The Go sample imports the library provided by the core Iter8 implementation. In addition to the API specific methods, some general gRPC libaries are required.
+
+=== "Node.js"
+    The manually generated node files [`abn_pd.js`](https://raw.githubusercontent.com/iter8-tools/docs/v0.13.0/samples/abn-sample/frontend/node/abn_pb.js) and [`abn_grpc_pb.js`](https://raw.githubusercontent.com/iter8-tools/docs/v0.13.0/samples/abn-sample/frontend/node/abn_grpc_pb.js) used used in the sample application can be copied and used without modification in other projects.
+
+    ```shell
+    var grpc = require('@grpc/grpc-js');
+
+    var messages = require('./abn_pb.js');
+    var services = require('./abn_grpc_pb.js');
+    ```
+
+=== "Go"
+    ```go
+    import (
+	    "google.golang.org/grpc"
+	    "google.golang.org/grpc/credentials/insecure"
+
+        pb "github.com/iter8-tools/iter8/abn/grpc"
+    )
+    ```
+
+### Instantiate a gRPC client
+
+Instantiating a client to the service implementing the SDK API is language specific.
+
+=== "Node.js"
+    ```javascript
+    var client = new services.ABNClient(abnEndpoint, grpc.credentials.createInsecure());
+    ```
+
+=== "Go"
+    ```go
+	opts := []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}
+	conn, err := grpc.Dial(fmt.Sprintf("%s:%s", getAbnService(), getAbnServicePort()), opts...)
+	if err != nil {
+		panic("Cannot establish connection with abn service")
+		// return
+	}
+	c := pb.NewABNClient(conn)
+	client = &c
+    ```
+
+### Define routing
+
+In the sample, track identifies are mapped to a static set of endpoints. The sample frontends maintain this in a simple map.
+
+=== "Node.js"
+    ```javascript
+    const trackToRoute = {
+        "backend":   "http://backend.default.svc.cluster.local:8091",
+        "backend-candidate-1": "http://backend-candidate-1.default.svc.cluster.local:8091",
+    }
+    ```
+
+=== "Go"
+    ```go
+    	trackToRoute = map[string]string{
+		"backend":             "http://backend.default.svc.cluster.local:8091",
+		"backend-candidate-1": "http://backend-candidate-1.default.svc.cluster.local:8091",
+	}
+    ```
```

Comment:
The JavaScript snippet is using 4 space spacing whereas the Go snippet is using single tab spacing. It would be nice if they match. This goes for all the code snippets.